graph/task.go

```diff
@@ -108,11 +118,17 @@ func NewTask(
 
 // initialize normalizes a Task's values.
 func (t *Task) initialize() error {
+	newDefaultNetworkName := DefaultNetworkName
+	// The default network being created is equal to the default network name passed in through CLI, if provided
+	addDefaultNetworkToSteps := false
+	if len(t.Networks) > 0 && t.Networks[len(t.Networks)-1].SkipCreation == true {
+		newDefaultNetworkName = t.Networks[len(t.Networks)-1].Name
+		addDefaultNetworkToSteps = true
+	}
 	// Add the default network if none are specified.
 	// Only add the default network if we're using tasks.
-	addDefaultNetworkToSteps := false
 	if !t.IsBuildTask && len(t.Networks) <= 0 {
-		defaultNetwork := NewNetwork(DefaultNetworkName, false, "bridge")
+		defaultNetwork := NewNetwork(DefaultNetworkName, false, "bridge", false)
```

Comment:
It actually doesn't matter since this block will be execute only when `len(t.Networks) <= 0`, that means the `newDefaultNetworkName` and `DefaultNetworkName` are the same. But this make sense to follow the consistency, changed to `DefaultNetworkName`.